sh_scrapy/extension.py

```diff
@@ -48,13 +49,24 @@ def spider_closed(self, spider, reason):
         self.pipe_writer.set_outcome(reason)
 
 
+_DOWNLOADER_MIDDLEWARE_WARNING = """\
+sh_scrapy.extension.HubstorageMiddleware functionality is now split between two new middlewares:
+
+- sh_scrapy.middlewares.HubstorageDownloaderMiddleware
+- sh_scrapy.middlewares.HubstorageSpiderMiddleware
+
+Please migrate to new middlewares.
+"""
+
+
 class HubstorageMiddleware(object):
 
     def __init__(self):
+        warnings.warn(_DOWNLOADER_MIDDLEWARE_WARNING)
```

Comment:
If this middleware is disabled by https://github.com/scrapinghub/scrapinghub-entrypoint-scrapy/pull/44/files#diff-8291937b18d088dbcac48282fe9f81d2R196, why don't just remove it?

New middlewares and old middleware are not compatible, they produce unexpected results, so it is not possible to use both at the same time. If someone is extending and/or manually enabling old middleware we should detect this case and do not enable new middlewares, or just warn and ignore old middleware.

A handy util to deprecate classes is https://github.com/scrapy/scrapy/blob/9731077a54978c375f386b6c156f9e74fe68cd03/scrapy/utils/deprecate.py#L15